HARK/tests/test_econforgeinterp.py

```diff
@@ -137,3 +138,125 @@ def test_metric(self):
         e_dist = distance_metric(LinearFast(z0, [x0, y0]), LinearFast(z1, [x1, y1]))
 
         self.assertAlmostEqual(h_dist, e_dist)
+
+
+# Derivative tests
+
+
+class Check1DDerivatives(unittest.TestCase):
+    """ 
+    Checks derivatives in a 1D interpolator
+    """
+
+    def setUp(self):
+
+        pass
+
+    def test_linear(self):
+
+        # A linear function on a non-uniform grid
+        x = np.exp(np.linspace(0, 2, 6))
+        y0 = 2
+        slope = 1.0
+        y = y0 + slope * x
+
+        interp = LinearFast(y, [x])
+
+        eval_points = np.array([1.5, 2.5, 3.5, 4.5])
+
+        grad = interp.gradient(eval_points)[0]
+        result = np.ones_like(eval_points) * slope
+
+        self.assertTrue(np.allclose(grad, result))
```

Comment:
It doesn't seem like the `derivative` method of `LinearInterp` is tested in the codebase. It might be useful to compare it to the `grad` output here to see if it's correct. Same for Bilinear, etc.

Reply:
Good idea, I've just added code to compare with the derivatives of `LinearInterp` and `BilinearInterp`